# RESUME SHORTLIST CLASSIFCATION
#Github link=https://github.com/harishnk28/ResumeShortlistPrediction

#Insight about the data:

The dataset contains information about the students resume of those who applied for the company.
I am going to predict the possibility of the students who has a chance of getting shortlisted based on certain category using the ML models.

In [ ]:


# importing the csv module
import csv

# my data rows as dictionary objects
mydict =[{'name':'harish','regno':'20aaa0009','branch':'cse','passoutyear':2022,'sex':'male','address':'4/169 vellore','mail':'harishnk28@gmail.com','cgpa':7.0,'skills':'c++','certification':'yes','project':'c++','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':60,'age':23,'requirement':'matched'},
         {'name':'babar','regno':'20aaa0033','branch':'cse','passoutyear':2022,'sex':'male','address':'9/619 madurai','mail':'bab00040@gmail.com','cgpa':7.5,'skills':'c++','certification':'no','project':'c++','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'lokchandar','regno':'20aaa0006','branch':'cse','passoutyear':2022,'sex':'male','address':'6/119 madurai','mail':'lokes0021@gmail.com','cgpa':7.0,'skills':'php','certification':'no','project':'angular','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'bhuvan','regno':'20aaa0099','branch':'cse','passoutyear':2022,'sex':'male','address':'5/339 madurai','mail':'bhuvan0023@gmail.com','cgpa':7.2,'skills':'.net','certification':'no','project':'.net','internorwexp':'yes','ppublication':'no','10th%':84,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'krithika','regno':'20aaa0058','branch':'cse','passoutyear':2022,'sex':'female','address':'8/929 tirunelveli','mail':'roh0803@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
        {'name':'kumar','regno':'20aaa0002','branch':'cse','passoutyear':2022,'sex':'male','address':'2/179 vellore','mail':'kumar0007@gmail.com','cgpa':6.0,'skills':'python','certification':'yes','project':'python','internorwexp':'no','ppublication':'no','10th%':60,'12th%':59,'age':24,'requirement':'notmatched'},
        {'name':'rajesh','regno':'20aaa0012','branch':'cse','passoutyear':2022,'sex':'male','address':'5/119 madurai','mail':'rajesh0003@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'suriya','regno':'20aaa0011','branch':'cse','passoutyear':2022,'sex':'male','address':'2/109 trichi','mail':'suriya0012@gmail.com','cgpa':6.9,'skills':'php','certification':'no','project':'php','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':21,'requirement':'notmatched'},
         {'name':'priya','regno':'20aaa0002','branch':'cse','passoutyear':2022,'sex':'female','address':'1/119 tirunelveli','mail':'priya0003@gmail.com','cgpa':8.2,'skills':'.net','certification':'yes','project':'NA','internorwexp':'yes','ppublication':'yes','10th%':59,'12th%':65,'age':23,'requirement':'notmatched'},
         {'name':'harini','regno':'20aaa0092','branch':'cse','passoutyear':2022,'sex':'female','address':'7/109 thanjavur','mail':'harini22@gmail.com','cgpa':7.0,'skills':'python','certification':'no','project':'python','internorwexp':'no','ppublication':'no','10th%':72,'12th%':75,'age':23,'requirement':'notmatched'},
         {'name':'kishore','regno':'20aaa0032','branch':'cse','passoutyear':2022,'sex':'male','address':'5/119 ariyalur','mail':'kishore0063@gmail.com','cgpa':7.2,'skills':'java','certification':'no','project':'angular','internorwexp':'no','ppublication':'no','10th%':66,'12th%':77,'age':24,'requirement':'notmatched'},
         {'name':'ramesh','regno':'20aaa010','branch':'cse','passoutyear':2022,'sex':'male','address':'2/144','mail':'ramesh003@gmail.com','cgpa':7.6,'skills':'mysql','certification':'yes','project':'python','internorwexp':'no','ppublication':'yes','10th%':80,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'suresh','regno':'20aaa0032','branch':'cse','passoutyear':2022,'sex':'male','address':'3/123 tiruvannamalai','mail':'suresh0012@gmail.com','cgpa':8.5,'skills':'sql','certification':'no','project':'sql','internorwexp':'yes','ppublication':'no','10th%':64,'12th%':60,'age':24,'requirement':'matched'},
         {'name':'shalini','regno':'20aaa0042','branch':'cse','passoutyear':2022,'sex':'female','address':'4/122 madurai','mail':'shalini0011@gmail.com','cgpa':7.0,'skills':'html','certification':'yes','project':'reactjs','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':23,'requirement':'notmatched'},
         {'name':'naveen','regno':'20aaa0032','branch':'cse','passoutyear':2022,'sex':'male','address':'7/129 vellore','mail':'naveen0013@gmail.com','cgpa':9.5,'skills':'java','certification':'no','project':'html','internorwexp':'yes','ppublication':'no','10th%':80,'12th%':87,'age':24,'requirement':'notmatched'},
         {'name':'supriya','regno':'20aaa0023','branch':'cse','passoutyear':2022,'sex':'female','address':'2/567 trichi','mail':'supria0008@gmail.com','cgpa':7.9,'skills':'nodejs','certification':'no','project':'nodejs','internorwexp':'yes','ppublication':'no','10th%':79,'12th%':65,'age':23,'requirement':'matched'},
         {'name':'kamal','regno':'20aaa0044','branch':'cse','passoutyear':2022,'sex':'male','address':'5/419 madurai','mail':'raj0007@gmail.com','cgpa':8.5,'skills':'mongodb','certification':'yes','project':'python','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'ganesh','regno':'20aaa0082','branch':'cse','passoutyear':2022,'sex':'male','address':'8/125 vellore','mail':'nirmal0035@gmail.com','cgpa':8.9,'skills':'angular','certification':'no','project':'angular','internorwexp':'no','ppublication':'yes','10th%':57,'12th%':65,'age':23,'requirement':'notmatched'},
         {'name':'nirmal','regno':'20aaa0062','branch':'cse','passoutyear':2022,'sex':'male','address':'5/119 madurai','mail':'rajesh0003@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'yes','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'dheepak','regno':'20aaa0052','branch':'cse','passoutyear':2022,'sex':'male','address':'3/452 madurai','mail':'kiresh34@gmail.com','cgpa':6.5,'skills':'python','certification':'no','project':'python','internorwexp':'no','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'govind','regno':'20aaa0092','branch':'cse','passoutyear':2022,'sex':'male','address':'7/110 vellore','mail':'govind0023@gmail.com','cgpa':7.5,'skills':'mongodb','certification':'no','project':'mongodb','internorwexp':'no','ppublication':'no','10th%':67,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'dina','regno':'20aaa0031','branch':'cse','passoutyear':2022,'sex':'male','address':'1/519 tiruchi','mail':'deen0006@gmail.com','cgpa':7.8,'skills':'java','certification':'no','project':'angular','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'keerthana','regno':'20aaa0068','branch':'cse','passoutyear':2022,'sex':'female','address':'9/189 vellore','mail':'keerthana0003@gmail.com','cgpa':9.5,'skills':'mysql','certification':'no','project':'angular','internorwexp':'yes','ppublication':'yes','10th%':60,'12th%':75,'age':24,'requirement':'matched'},
         {'name':'yogaraj','regno':'20aaa0074','branch':'cse','passoutyear':2022,'sex':'male','address':'5/178 thanjavur','mail':'yoga0045@gmail.com','cgpa':7.4,'skills':'c++','certification':'yes','project':'python','internorwexp':'no','ppublication':'yes','10th%':70,'12th%':78,'age':23,'requirement':'matched'},
         {'name':'karthik','regno':'20aaa0055','branch':'cse','passoutyear':2022,'sex':'male','address':'6/132 vellore','mail':'karthik0006@gmail.com','cgpa':7.2,'skills':'angular','certification':'no','project':'angular','internorwexp':'no','ppublication':'yes','10th%':58,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'babu','regno':'20aaa0093','branch':'cse','passoutyear':2022,'sex':'male','address':'5/619 madurai','mail':'babu0045@gmail.com','cgpa':7.5,'skills':'c++','certification':'yes','project':'c++','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'lokesh','regno':'20aaa0016','branch':'cse','passoutyear':2022,'sex':'male','address':'9/119 madurai','mail':'lokesh0001@gmail.com','cgpa':7.0,'skills':'php','certification':'no','project':'angular','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'bhuvanesh','regno':'20aaa0019','branch':'cse','passoutyear':2022,'sex':'male','address':'5/339 madurai','mail':'bhuvan0023@gmail.com','cgpa':7.2,'skills':'.net','certification':'no','project':'.net','internorwexp':'yes','ppublication':'no','10th%':84,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'rohini','regno':'20aaa0018','branch':'cse','passoutyear':2022,'sex':'female','address':'8/219 tirunelveli','mail':'rohini0003@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'no','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'geetha','regno':'20aaa0005','branch':'cse','passoutyear':2022,'sex':'female','address':'7/610 vellore','mail':'geetha0023@gmail.com','cgpa':6.8,'skills':'php','certification':'no','project':'mongodb','internorwexp':'yes','ppublication':'no','10th%':89,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'aravind','regno':'20aaa0041','branch':'cse','passoutyear':2022,'sex':'male','address':'1/549 tiruchi','mail':'aravind0006@gmail.com','cgpa':7.8,'skills':'java','certification':'no','project':'angular','internorwexp':'no','ppublication':'no','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'keerthi','regno':'20aaa0048','branch':'cse','passoutyear':2022,'sex':'female','address':'2/189 vellore','mail':'keerthi0003@gmail.com','cgpa':9.5,'skills':'mysql','certification':'no','project':'mysql','internorwexp':'no','ppublication':'yes','10th%':60,'12th%':75,'age':24,'requirement':'matched'},
         {'name':'yogaraj','regno':'20aaa0059','branch':'cse','passoutyear':2022,'sex':'male','address':'9/178 thanjavur','mail':'yogaraj0045@gmail.com','cgpa':7.4,'skills':'c++','certification':'yes','project':'python','internorwexp':'yes','ppublication':'no','10th%':70,'12th%':78,'age':23,'requirement':'matched'},
         {'name':'karthik','regno':'20aaa0060','branch':'cse','passoutyear':2022,'sex':'male','address':'6/132 vellore','mail':'karthik0006@gmail.com','cgpa':7.2,'skills':'angular','certification':'no','project':'angular','internorwexp':'no','ppublication':'no','10th%':58,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'babu','regno':'20aaa0043','branch':'cse','passoutyear':2022,'sex':'male','address':'4/619 madurai','mail':'bab00045@gmail.com','cgpa':7.5,'skills':'c++','certification':'yes','project':'c++','internorwexp':'yes','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'lokesh','regno':'20aaa0086','branch':'cse','passoutyear':2022,'sex':'male','address':'3/119 madurai','mail':'lokes0001@gmail.com','cgpa':7.0,'skills':'php','certification':'no','project':'angular','internorwexp':'no','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'notmatched'},
         {'name':'bhuvanesh','regno':'20aaa0039','branch':'cse','passoutyear':2022,'sex':'male','address':'5/339 madurai','mail':'bhuvan0023@gmail.com','cgpa':7.2,'skills':'.net','certification':'no','project':'.net','internorwexp':'yes','ppublication':'no','10th%':84,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'rohini','regno':'20aaa0098','branch':'cse','passoutyear':2022,'sex':'female','address':'8/719 tirunelveli','mail':'rohini8003@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'yes','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'jack','regno':'20aaa0030','branch':'cse','passoutyear':2022,'sex':'male','address':'1/132 vellore','mail':'erthik0006@gmail.com','cgpa':7.2,'skills':'angular','certification':'yes','project':'java','internorwexp':'yes','ppublication':'yes','10th%':78,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'bablu','regno':'20aaa0093','branch':'cse','passoutyear':2022,'sex':'male','address':'4/619 madurai','mail':'bab00045@gmail.com','cgpa':7.5,'skills':'c++','certification':'no','project':'c++','internorwexp':'no','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'lokchand','regno':'20aaa0066','branch':'cse','passoutyear':2022,'sex':'male','address':'3/119 madurai','mail':'lokes0001@gmail.com','cgpa':7.0,'skills':'php','certification':'no','project':'angular','internorwexp':'yes','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'},
         {'name':'bhuvi','regno':'20aaa0079','branch':'cse','passoutyear':2022,'sex':'male','address':'5/339 madurai','mail':'bhuvan0023@gmail.com','cgpa':7.2,'skills':'.net','certification':'no','project':'.net','internorwexp':'yes','ppublication':'no','10th%':84,'12th%':85,'age':24,'requirement':'notmatched'},
         {'name':'krithika','regno':'20aaa0088','branch':'cse','passoutyear':2022,'sex':'female','address':'8/919 tirunelveli','mail':'roh8803@gmail.com','cgpa':7.5,'skills':'angular','certification':'no','project':'angular','internorwexp':'yes','ppublication':'yes','10th%':60,'12th%':65,'age':24,'requirement':'matched'}
        ]
# field names
fields = ['name','regno','branch', 'passoutyear','sex','address','mail','cgpa','skills','certification','project','internorwexp','ppublication','10th%','12th%','age','requirement']
# name of csv file
filename = "harish_paccount.csv"

# writing to csv file
with open(filename, 'w') as csvfile:
    # creating a csv dict writer object
    writer = csv.DictWriter(csvfile, fieldnames = fields)
    # writing headers (field names)
    writer.writeheader()
    writer.writerows(mydict)

In [126]:
#Reading csv file
import pandas as pd
dataset=pd.read_csv('harish_paccount.csv')
dataset

name      regno branch  passoutyear     sex               address  \
0       harish  20aaa0009    cse         2022    male         4/169 vellore   
1        babar  20aaa0033    cse         2022    male         9/619 madurai   
2   lokchandar  20aaa0006    cse         2022    male         6/119 madurai   
3       bhuvan  20aaa0099    cse         2022    male         5/339 madurai   
4     krithika  20aaa0058    cse         2022  female     8/929 tirunelveli   
5        kumar  20aaa0002    cse         2022    male         2/179 vellore   
6       rajesh  20aaa0012    cse         2022    male         5/119 madurai   
7       suriya  20aaa0011    cse         2022    male          2/109 trichi   
8        priya  20aaa0002    cse         2022  female     1/119 tirunelveli   
9       harini  20aaa0092    cse         2022  female       7/109 thanjavur   
10     kishore  20aaa0032    cse         2022    male        5/119 ariyalur   
11      ramesh   20aaa010    cse         2022    male                 2/144   
12      suresh  20aaa0032    cse         2022    male  3/123 tiruvannamalai   
13     shalini  20aaa0042    cse         2022  female         4/122 madurai   
14      naveen  20aaa0032    cse         2022    male         7/129 vellore   
15     supriya  20aaa0023    cse         2022  female          2/567 trichi   
16       kamal  20aaa0044    cse         2022    male         5/419 madurai   
17      ganesh  20aaa0082    cse         2022    male         8/125 vellore   
18      nirmal  20aaa0062    cse         2022    male         5/119 madurai   
19     dheepak  20aaa0052    cse         2022    male         3/452 madurai   
20      govind  20aaa0092    cse         2022    male         7/110 vellore   
21        dina  20aaa0031    cse         2022    male         1/519 tiruchi   
22   keerthana  20aaa0068    cse         2022  female         9/189 vellore   
23     yogaraj  20aaa0074    cse         2022    male       5/178 thanjavur   
24     karthik  20aaa0055    cse         2022    male         6/132 vellore   
25        babu  20aaa0093    cse         2022    male         5/619 madurai   
26      lokesh  20aaa0016    cse         2022    male         9/119 madurai   
27   bhuvanesh  20aaa0019    cse         2022    male         5/339 madurai   
28      rohini  20aaa0018    cse         2022  female     8/219 tirunelveli   
29      geetha  20aaa0005    cse         2022  female         7/610 vellore   
30     aravind  20aaa0041    cse         2022    male         1/549 tiruchi   
31     keerthi  20aaa0048    cse         2022  female         2/189 vellore   
32     yogaraj  20aaa0059    cse         2022    male       9/178 thanjavur   
33     karthik  20aaa0060    cse         2022    male         6/132 vellore   
34        babu  20aaa0043    cse         2022    male         4/619 madurai   
35      lokesh  20aaa0086    cse         2022    male         3/119 madurai   
36   bhuvanesh  20aaa0039    cse         2022    male         5/339 madurai   
37      rohini  20aaa0098    cse         2022  female     8/719 tirunelveli   
38        jack  20aaa0030    cse         2022    male         1/132 vellore   
39       bablu  20aaa0093    cse         2022    male         4/619 madurai   
40    lokchand  20aaa0066    cse         2022    male         3/119 madurai   
41       bhuvi  20aaa0079    cse         2022    male         5/339 madurai   
42    krithika  20aaa0088    cse         2022  female     8/919 tirunelveli   

                       mail  cgpa   skills certification  project  \
0      harishnk28@gmail.com   7.0      c++           yes      c++   
1        bab00040@gmail.com   7.5      c++            no      c++   
2       lokes0021@gmail.com   7.0      php            no  angular   
3      bhuvan0023@gmail.com   7.2     .net            no     .net   
4         roh0803@gmail.com   7.5  angular            no  angular   
5       kumar0007@gmail.com   6.0   python           yes   python   
6      rajesh0003@gmail.com   7.5  angular        

In [127]:
#Null  count
print(" \nCount total NaN at each column in a DataFrame : \n\n", 
      dataset.isnull().sum())

 
Count total NaN at each column in a DataFrame : 

 name             0
regno            0
branch           0
passoutyear      0
sex              0
address          0
mail             0
cgpa             0
skills           0
certification    0
project          1
internorwexp     0
ppublication     0
10th%            0
12th%            0
age              0
requirement      0
dtype: int64


In [128]:
#Dataset Info
dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 43 entries, 0 to 42
Data columns (total 17 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   name           43 non-null     object 
 1   regno          43 non-null     object 
 2   branch         43 non-null     object 
 3   passoutyear    43 non-null     int64  
 4   sex            43 non-null     object 
 5   address        43 non-null     object 
 6   mail           43 non-null     object 
 7   cgpa           43 non-null     float64
 8   skills         43 non-null     object 
 9   certification  43 non-null     object 
 10  project        42 non-null     object 
 11  internorwexp   43 non-null     object 
 12  ppublication   43 non-null     object 
 13  10th%          43 non-null     int64  
 14  12th%          43 non-null     int64  
 15  age            43 non-null     int64  
 16  requirement    43 non-null     object 
dtypes: float64(1), int64(4), object(12)
memory usage: 5.8+ K

In [129]:
#Rename columns
dataset = dataset.rename({'requirement': 'target','10th%':'tenthscore','12th%':'twelfthscore'},axis=1)

In [130]:
#Choosing the Predictor variable or independent variable
predictor=dataset[['cgpa','skills','certification','project','internorwexp','ppublication','tenthscore','twelfthscore','age']]
predictor

cgpa   skills certification  project internorwexp ppublication  \
0    7.0      c++           yes      c++          yes           no   
1    7.5      c++            no      c++          yes           no   
2    7.0      php            no  angular          yes           no   
3    7.2     .net            no     .net          yes           no   
4    7.5  angular            no  angular          yes           no   
5    6.0   python           yes   python           no           no   
6    7.5  angular            no  angular          yes           no   
7    6.9      php            no      php           no           no   
8    8.2     .net           yes      NaN          yes          yes   
9    7.0   python            no   python           no           no   
10   7.2     java            no  angular           no           no   
11   7.6    mysql           yes   python           no          yes   
12   8.5      sql            no      sql          yes           no   
13   7.0     html           yes  reactjs           no           no   
14   9.5     java            no     html          yes           no   
15   7.9   nodejs            no   nodejs          yes           no   
16   8.5  mongodb           yes   python           no           no   
17   8.9  angular            no  angular           no          yes   
18   7.5  angular            no  angular          yes           no   
19   6.5   python            no   python           no          yes   
20   7.5  mongodb            no  mongodb           no           no   
21   7.8     java            no  angular           no           no   
22   9.5    mysql            no  angular          yes          yes   
23   7.4      c++           yes   python           no          yes   
24   7.2  angular            no  angular           no          yes   
25   7.5      c++           yes      c++           no           no   
26   7.0      php            no  angular           no           no   
27   7.2     .net            no     .net          yes           no   
28   7.5  angular            no  angular           no          yes   
29   6.8      php            no  mongodb          yes           no   
30   7.8     java            no  angular           no           no   
31   9.5    mysql            no    mysql           no          yes   
32   7.4      c++           yes   python          yes           no   
33   7.2  angular            no  angular           no           no   
34   7.5      c++           yes      c++          yes          yes   
35   7.0      php            no  angular           no          yes   
36   7.2     .net            no     .net          yes           no   
37   7.5  angular            no  angular          yes          yes   
38   7.2  angular           yes     java          yes          yes   
39   7.5      c++            no      c++           no          yes   
40   7.0      php            no  angular          yes          yes   
41   7.2     .net            no     .net          yes           no   
42   7.5  angular            no  angular          yes          yes   

    tenthscore  twelfthscore  age  
0           60            60   23  
1           60            65   24  
2           60            65   24  
3           84            85   24  
4           60            65   24  
5           60            59   24  
6           60            65   24  
7           60            65   21  
8           59            65   23  
9           72            75   23  
10          66            77   24  
11          80            65   24  
12          64            60   24  
13          60            65   23  
14          80            87   24  
15          79            65   23  
16          60            65   24  
17          57            65   23  
18          60            65   24  
19          60            65   24  
20          67            85   24  
21          60            65   24  
22          60            75   24  
23          70            78   23  
24          58         

In [131]:
#choosing the target variable or dependent variable
y=dataset[['target']]
y

target
0      matched
1      matched
2   notmatched
3   notmatched
4      matched
5   notmatched
6      matched
7   notmatched
8   notmatched
9   notmatched
10  notmatched
11     matched
12     matched
13  notmatched
14  notmatched
15     matched
16  notmatched
17  notmatched
18     matched
19  notmatched
20  notmatched
21  notmatched
22     matched
23     matched
24  notmatched
25     matched
26  notmatched
27  notmatched
28     matched
29  notmatched
30  notmatched
31     matched
32     matched
33  notmatched
34     matched
35  notmatched
36  notmatched
37     matched
38     matched
39     matched
40     matched
41  notmatched
42     matched

In [132]:
#encoding the matched values to 1 else 0
y=y["target"].apply(lambda target:1 if target=='matched' else 0)
y

0     1
1     1
2     0
3     0
4     1
5     0
6     1
7     0
8     0
9     0
10    0
11    1
12    1
13    0
14    0
15    1
16    0
17    0
18    1
19    0
20    0
21    0
22    1
23    1
24    0
25    1
26    0
27    0
28    1
29    0
30    0
31    1
32    1
33    0
34    1
35    0
36    0
37    1
38    1
39    1
40    1
41    0
42    1
Name: target, dtype: int64

In [133]:
#Choosing the required skills
predictor["skills"]=predictor["skills"].apply(lambda skills:skills if skills=="java" or skills=='python' or skills=='angular' or skills=='c++' or skills=='php' or skills=='sql' or skills=='mongodb' or skills=='reactjs' or skills=='html' or skills=='mysql' else 0)

#Extracting  the skills and project matched values
import numpy as np
predictor['skillprojmatch'] = np.where(predictor['skills'] == predictor['project'], 1, 0)
predictor


<ipython-input-133-1d59cc3da40b>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  predictor["skills"]=predictor["skills"].apply(lambda skills:skills if skills=="java" or skills=='python' or skills=='angular' or skills=='c++' or skills=='php' or skills=='sql' or skills=='mongodb' or skills=='reactjs' or skills=='html' or skills=='mysql' else 0)
<ipython-input-133-1d59cc3da40b>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  predictor['skillprojmatch'] = np.where(predictor['skills'] == predictor['project'], 

cgpa   skills certification  project internorwexp ppublication  \
0    7.0      c++           yes      c++          yes           no   
1    7.5      c++            no      c++          yes           no   
2    7.0      php            no  angular          yes           no   
3    7.2        0            no     .net          yes           no   
4    7.5  angular            no  angular          yes           no   
5    6.0   python           yes   python           no           no   
6    7.5  angular            no  angular          yes           no   
7    6.9      php            no      php           no           no   
8    8.2        0           yes      NaN          yes          yes   
9    7.0   python            no   python           no           no   
10   7.2     java            no  angular           no           no   
11   7.6    mysql           yes   python           no          yes   
12   8.5      sql            no      sql          yes           no   
13   7.0     html           yes  reactjs           no           no   
14   9.5     java            no     html          yes           no   
15   7.9        0            no   nodejs          yes           no   
16   8.5  mongodb           yes   python           no           no   
17   8.9  angular            no  angular           no          yes   
18   7.5  angular            no  angular          yes           no   
19   6.5   python            no   python           no          yes   
20   7.5  mongodb            no  mongodb           no           no   
21   7.8     java            no  angular           no           no   
22   9.5    mysql            no  angular          yes          yes   
23   7.4      c++           yes   python           no          yes   
24   7.2  angular            no  angular           no          yes   
25   7.5      c++           yes      c++           no           no   
26   7.0      php            no  angular           no           no   
27   7.2        0            no     .net          yes           no   
28   7.5  angular            no  angular           no          yes   
29   6.8      php            no  mongodb          yes           no   
30   7.8     java            no  angular           no           no   
31   9.5    mysql            no    mysql           no          yes   
32   7.4      c++           yes   python          yes           no   
33   7.2  angular            no  angular           no           no   
34   7.5      c++           yes      c++          yes          yes   
35   7.0      php            no  angular           no          yes   
36   7.2        0            no     .net          yes           no   
37   7.5  angular            no  angular          yes          yes   
38   7.2  angular           yes     java          yes          yes   
39   7.5      c++            no      c++           no          yes   
40   7.0      php            no  angular          yes          yes   
41   7.2        0            no     .net          yes           no   
42   7.5  angular            no  angular          yes          yes   

    tenthscore  twelfthscore  age  skillprojmatch  
0           60            60   23               1  
1           60            65   24               1  
2           60            65   24               0  
3           84            85   24               0  
4           60            65   24               1  
5           60            59   24               1  
6           60            65   24               1  
7           60            65   21               1  
8           59            65   23               0  
9           72            75   23               1  
10          66            77   24               0  
11          80            65   24               0  
12          64            60   24               1  
13          60            65   23               0  
14          80            87   24               0  
15          79            65   23               0  
16          60            65   24      

In [134]:
#Manual Binary convertion

#predictor["skills"]=predictor["skills"].apply(lambda skills:1 if skills=="java" or skills=='python' or skills=='angular' or skills=='c++' or skills=='php' or skills=='sql' or skills=='mongodb' or skills=='reactjs' or skills=='html' or skills=='mysql' else 0)
predictor["certification"]=predictor["certification"].apply(lambda certification:1 if certification=='yes' else 0)
predictor["internorwexp"]=predictor["internorwexp"].apply(lambda internorwexp:1 if internorwexp=='yes' else 0)
predictor["ppublication"]=predictor["ppublication"].apply(lambda ppublication:1 if ppublication=='yes' else 0)
#predictor["project"]=predictor["project"].apply(lambda project:1 if project=="java" or project=='python' or project=='angular' or project=='c++' or project=='php' or project=='sql' or project=='mongodb' or project=='reactjs' or project=='html' or project=='mysql' else 0)
predictor["tenthscore"]=predictor["tenthscore"].apply(lambda tenthscore:1 if tenthscore>59 else 0)
predictor["twelfthscore"]=predictor["twelfthscore"].apply(lambda twelfthscore:1 if twelfthscore>59 else 0)
predictor["age"]=predictor["age"].apply(lambda age:1 if age==24 or age==23 else 0)
predictor["cgpa"]=predictor["cgpa"].apply(lambda cgpa:1 if cgpa>=7.0 else 0)

<ipython-input-134-7d32241173ae>:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  predictor["certification"]=predictor["certification"].apply(lambda certification:1 if certification=='yes' else 0)
<ipython-input-134-7d32241173ae>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  predictor["internorwexp"]=predictor["internorwexp"].apply(lambda internorwexp:1 if internorwexp=='yes' else 0)
<ipython-input-134-7d32241173ae>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataF

In [135]:
#Filtering out the eligible category students

predictor=pd.concat([predictor,y],axis=1)
predictor = predictor[predictor['cgpa'] == 1]
predictor=predictor[predictor['tenthscore']==1]
predictor=predictor[predictor['twelfthscore']==1]
predictor=predictor[predictor['age']==1]
predictor

cgpa   skills  certification  project  internorwexp  ppublication  \
0      1      c++              1      c++             1             0   
1      1      c++              0      c++             1             0   
2      1      php              0  angular             1             0   
3      1        0              0     .net             1             0   
4      1  angular              0  angular             1             0   
6      1  angular              0  angular             1             0   
9      1   python              0   python             0             0   
10     1     java              0  angular             0             0   
11     1    mysql              1   python             0             1   
12     1      sql              0      sql             1             0   
13     1     html              1  reactjs             0             0   
14     1     java              0     html             1             0   
15     1        0              0   nodejs             1             0   
16     1  mongodb              1   python             0             0   
18     1  angular              0  angular             1             0   
20     1  mongodb              0  mongodb             0             0   
21     1     java              0  angular             0             0   
22     1    mysql              0  angular             1             1   
23     1      c++              1   python             0             1   
25     1      c++              1      c++             0             0   
26     1      php              0  angular             0             0   
27     1        0              0     .net             1             0   
28     1  angular              0  angular             0             1   
30     1     java              0  angular             0             0   
31     1    mysql              0    mysql             0             1   
32     1      c++              1   python             1             0   
34     1      c++              1      c++             1             1   
35     1      php              0  angular             0             1   
36     1        0              0     .net             1             0   
37     1  angular              0  angular             1             1   
38     1  angular              1     java             1             1   
39     1      c++              0      c++             0             1   
40     1      php              0  angular             1             1   
41     1        0              0     .net             1             0   
42     1  angular              0  angular             1             1   

    tenthscore  twelfthscore  age  skillprojmatch  target  
0            1             1    1               1       1  
1            1             1    1               1       1  
2            1             1    1               0       0  
3            1             1    1               0       0  
4            1             1    1               1       1  
6            1             1    1               1       1  
9            1             1    1               1       0  
10           1             1    1               0       0  
11           1             1    1               0       1  
12           1             1    1               1       1  
13           1             1    1               0       0  
14           1             1    1               0       0  
15           1             1    1               0       1  
16           1             1    1               0       0  
18           1             1    1               1       1  
20           1             1    1               1       0  
21           1             1    1               0       0  
22           1             1    1               0       1  
23           1             1    1               0       1  
25           1             1    1               1       1  
26           1             1    1               0       0  
27           1             1    1               0      

In [136]:
#finalizing the predictor variable by dropping unnecessary values

y=predictor['target']
predictor=predictor.drop(['skills','project','age','target'],axis=1)
print(predictor)
print(y)

    cgpa  certification  internorwexp  ppublication  tenthscore  twelfthscore  \
0      1              1             1             0           1             1   
1      1              0             1             0           1             1   
2      1              0             1             0           1             1   
3      1              0             1             0           1             1   
4      1              0             1             0           1             1   
6      1              0             1             0           1             1   
9      1              0             0             0           1             1   
10     1              0             0             0           1             1   
11     1              1             0             1           1             1   
12     1              0             1             0           1             1   
13     1              1             0             0           1             1   
14     1              0     

In [137]:
#splitting dataset into Train and Test set
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(predictor,y,test_size=0.3,random_state=15)


In [138]:
#call the Logistic regression model
from sklearn.linear_model import LogisticRegression
logit=LogisticRegression()
logit.fit(x_train,y_train)

LogisticRegression()

In [139]:
#computing the predictions
y_pred=logit.predict(x_test)

In [140]:
# LOGISTIC REGRESSION confusion matrix
from sklearn.metrics import confusion_matrix
cf=confusion_matrix(y_test,y_pred)
cf

array([[3, 0],
       [2, 6]], dtype=int64)

In [141]:
# LOGISTIC REGRESSION accuracy score
from sklearn.metrics import accuracy_score
accuracy_score(y_test,y_pred)

0.8181818181818182

In [142]:
# LOGISTIC REGRESSION classification report
from sklearn.metrics import classification_report
report=classification_report(y_test,y_pred)
print(report)

              precision    recall  f1-score   support

           0       0.60      1.00      0.75         3
           1       1.00      0.75      0.86         8

    accuracy                           0.82        11
   macro avg       0.80      0.88      0.80        11
weighted avg       0.89      0.82      0.83        11



In [143]:
#NAIVE BAYES model
from sklearn.naive_bayes import GaussianNB
gn=GaussianNB()
gn.fit(x_train,y_train)

GaussianNB()

In [144]:
#Testing the model with single test value
ypredt=gn.predict([[1,0,1,0,1,1,0]])
if ypredt[0]==1:
    print("shortlisted")
else:
    print("not shortlisted")

not shortlisted


In [145]:
ypred=gn.predict(x_test)

In [146]:
#NAIVE BAYES confusion matrix
from sklearn.metrics import confusion_matrix
cf1=confusion_matrix(y_test,ypred)
cf1

array([[3, 0],
       [1, 7]], dtype=int64)

In [147]:
# NAIVE BAYES accuracy score
from sklearn.metrics import accuracy_score
accuracy_score(y_test,ypred)

0.9090909090909091

In [148]:
#NAIVE BAYES classification report
from sklearn.metrics import classification_report
report1=classification_report(y_test,ypred)
print(report1)

              precision    recall  f1-score   support

           0       0.75      1.00      0.86         3
           1       1.00      0.88      0.93         8

    accuracy                           0.91        11
   macro avg       0.88      0.94      0.90        11
weighted avg       0.93      0.91      0.91        11



Conclusion:-
Thus,we can able to see that there is 9 percent difference in the prediction by the models
LOGISTIC REGRESSION ACCURACY SCORE = 0.8181818181818182
NAIVE BAYES ACCURACY SCORE         = 0.9090909090909091

NAIVE BAYES PERFORMS BETTER THAN LOGISTIC REGRESSION FOR THE GIVEN DATASET